cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -334,3 +353,160 @@ func _execute_code{
     // Recursive call to continue execution
     return _execute_code(evm);
 }
+
+func process_message_call{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(message: Message, env: Environment) -> (MessageCallOutput, EthereumException*) {
```

Comment:
take env implicitly, because it contains the state, which is mutated during the transaction. before returning from this function, you need to rebind the `env` variable to its latest version (from evm.env - if not an error.)